File: Practice/Linked list/llist.js
class Node{
    constructor(value){
        this.value = value
        this.next = null
        this.prev = null
    }
}
class LinkedList{
    constructor(){
        this.head = null
        this.tail = null
        this.size = 0
    }
    isEmpty(){
        return this.size === 0;
    }

}

const llist = new LinkedList()
console.log(llist.size)

File: Practice/Linked list/llist.cpp
#include<iostream>
using namespace std;
class Node{
public:
    int value;
    Node* next;
    Node* prev;
    Node(int value){
        this -> value = value;
        this -> next = nullptr; 
    };
};

class LinkedList{
private:
    Node* head;
    Node* tail;
public:
    int size;
    LinkedList(){
        this -> head = nullptr;
        this -> tail = nullptr;
        this -> size = 0;
    }
    bool isEmpty(){
        return this -> size == 0;
    }
    void prepend(int value){
        Node* node = new Node(value);
        if(this -> isEmpty()){
            this -> head = node;
            this -> tail = node;
        } else{
            this->head ->prev = node;
            node -> next = this -> head;
            this -> head = node;
        }
        this -> size++;
    }
};

int main(){
    LinkedList llist = LinkedList();
    cout << llist.isEmpty() << endl;
    return 0;
}